### PR TITLE
MINOR: Improve Docker examples readme

### PR DESCRIPTION
**Pull request by fvaleri**
This patch adds some clarity about the environment variables usage and limitations. It also removes the invalid example and adds few style fixes.

See https://github.com/apache/kafka/pull/17705#issuecomment-2488517083.

### CODE REVIEW

**Comment by fvaleri** [comment]
@VedarthConfluent @kagarwal06 fyi

**Comment by fvaleri** [comment]
@showuon @mimaison fyi

File: docker/examples/README.md [hunk 1]
```diff
@@ -4,58 +4,61 @@ Kafka Docker Image Usage Guide
 Introduction
 ------------
 
-This document contains usage guide as well as examples for docker image. Docker compose files are provided in this directory for the example use cases.
+This document contains usage guide as well as examples for Docker image.
+Docker Compose files are provided in this directory for the example use cases.
+
+Kafka server can be started using following ways:
 
-Kafka server can be started using following ways:-
```
Comment (showuon) [inline docker/examples/README.md:9]: Nice catch!

File: docker/examples/README.md [hunk 2]
```diff
@@ -4,58 +4,61 @@ Kafka Docker Image Usage Guide
...
 - File input
 - Environment variables
 
 Running on default configs
 --------------------------
 
-If no user provided configs are passed to docker container or configs provided are empty, default configs will be used (configs that are packaged in kafka tarball). If any user provided config is provided, default configs will not be used.
+If no user provided configuration file is passed to the Docker container, or the file is empty, the default KRaft configuration for single combined-mode node will be used.
+This default configuration is packaged in the Kafka tarball.
+If any user provided config (file input or environment variables) is provided, the default configuration will not be used.
```
Comment (showuon) [inline docker/examples/README.md]: I think we can combine L19 and L21 to sentence.

File: docker/examples/README.md [hunk 3]
```diff
@@ -4,58 +4,61 @@ Kafka Docker Image Usage Guide
...
 ---------------------------
 
-- Kafka property defined via env variables will override the value of that property defined in file input and default config.
-- If properties are provided via environment variables only, default configs will be replaced by user provided properties.
-- To construct the environment key variable name for server.properties configs, following steps can be followed:-
-    - Replace . with _
-    - Replace _ with __(double underscore)
-    - Replace - with ___(triple underscore)
-    - Prefix the result with KAFKA_
+One limitation of the environment variables approach it that you need to set all properties required to start the KRaft node.
```
Comment (showuon) [inline docker/examples/README.md]: I don't think we should call it as "limitation", maybe:
`Note: when using the environment variables approach it that you need to set all properties required to start the KRaft node.`

**Review by showuon — COMMENTED** [review]
Thanks for the PR. Left some comments.

**Comment by showuon** [comment]
@VedarthConfluent @kagarwal06, call for review. Thanks.

**Review by VedarthConfluent — APPROVED** [review]
Thanks for updating the docs for better clarity! LGTM.

File: docker/examples/README.md (hunk 3 above)
Comment (VedarthConfluent) [reply docker/examples/README.md]: Agreed

Comment (fvaleri) [reply docker/examples/README.md]: OK.

**Comment by mimaison** [comment]
Why are we keeping this information in a README, shouldn't this be directly part of the documentation?

**Review by showuon — APPROVED** [review]
LGTM!

**Comment by showuon** [comment]
> Why are we keeping this information in a README, shouldn't this be directly part of the documentation?

+1. @fvaleri , could you help create a JIRA and if you have time, adding the content in README to the doc? 
Thank you.

**Pull request merged**